<html>
<body>
報告對 <code>Throwable.initCause()</code> 的呼叫，其中異常建構函式還包含 <code>Throwable cause</code> 實參。
<p>在本例中，可以移除 <code>initCause()</code> 呼叫，並將其實參新增到對異常建構函式的呼叫中。</p>
<p><b>範例：</b></p>
<pre><code>
  try {
      process();
  }
  catch (RuntimeException ex) {
    RuntimeException wrapper = new RuntimeException("Error while processing");
    wrapper.initCause(ex); // 不必要呼叫 'Throwable.initCause()'
    throw wrapper;
  }
</code></pre>
<p>可通過快速修復將 cause 實參傳遞給建構函式。 在套用快速修復後：</p>
<pre><code>
  try {
      process();
  }
  catch (RuntimeException ex) {
    RuntimeException wrapper = new RuntimeException("Error while processing", ex);
    throw wrapper;
  }
  </code></pre>
<!-- tooltip end -->
</body>
</html>